File: Results/DER/Translation/translation_output/w_test/gpt-4/codenet/python/java-sanitized/p03583.java
import java.util.Scanner;

public class p03583 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        scanner.close();
        
        for (int i = 1; i <= 3500; i++) {
            for (int j = 1; j <= 3500; j++) {
                long tmp1 = (long) N * i * j;
                long tmp2 = 4L * i * j - (long) N * j - (long) N * i;
                
                if (tmp2 != 0) {
                    long s = tmp1 / tmp2;
                    long m = tmp1 % tmp2;
                    
                    if (m == 0 && s >= 0) {
                        System.out.println(i + " " + j + " " + s);
                        return;
                    }
                }
            }
        }
    }
}

File: Results/DER/Translation/translation_output/no_test/llama2/codenet/java/python/p03583.py
<s> Translate the following java code to python and enclose your solution inside ```python```:
```
import java.util.Scanner;

class p03583 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        sc.close();
        // a
        for (int i = 1; i < 3510; i++) {
            // b
            for (int j = 1; j < 3510; j++) {
                long si = n * i * j;
                long bo = 4 * i * j - n * j - n * i;
                if (bo <= 0) {
                    continue;
                }
                if (si % bo == 0) {
                    System.out.println(i + " " + j + " " + (si / bo));
                    return;
                }
            }
        }
    }
}

```

### 解题思路

```python
import math

def main():
    n = int(input())
    for i in range(1, 3510):
        for j in range(1, 3510):
            si = n * i * j
            bo = 4 * i * j - n * j - n * i
            if bo <= 0:
                continue
            if si % bo == 0:
                print(i, j, si // bo)
                return

if __name__ == '__main__':
    main()
```

### 复杂度分析

- 时间复杂度：O(n^2)。
- 空间复杂度：O(1)。

### 相关题目

- [p03582](https://leetcode-cn.com/problems/p03582/)
</s>
